homeassistant/components/varta_storage/const.py

```diff
@@ -0,0 +1,190 @@
+"""Constants for the VARTA Storage integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+from typing import Final
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import ENERGY_WATT_HOUR, PERCENTAGE, POWER_WATT, TIME_HOURS
+
+DOMAIN = "varta_storage"
+LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=1)
```

Comment:
Is it reasonable to query the current state every second? Or am I overlooking something?

Reply:
I thought about that. I understand that it would be a good practice to avoid unnecessary fast scan intervals.
Initially i had this set to a higher value of 10. However in my case is use this state values to control a heating element based on the measured oversupply by the battery. For this usecase its good to have up2data data to avoid power consumption from the grid.

Do you see a problem if we let it like that? I also can imagine to make this somehow "configurable" to the user and set the default to a higher value. Would that be a approach you can agree on?

Reply:
I think that is already configurable: https://www.home-assistant.io/docs/configuration/platform_options/#scan-interval

But I've never tried that in production

Reply:
tried it using this in the configuration.yaml on my testsystem. 

`sensor:
  - platform: integration
    scan_interval: 1
    source: sensor.varta_grid_power`

Expected behaviour would be that this sensor gets polled every second instead of the 10 seconds i configured for the integration itself. Unfortunately this is not working like that :(

Reply:
Personally, I would've used a configuration as follows (untested):
```yaml
varta_storage:
  - platform: sensor
    scan_interval: 1
```

Reply:
Personally, I would've used a configuration as follows (untested):
```yaml
sensor:
  - platform: varta_storage
    scan_interval: 1
```

Reply:
A different approach would be an automation for updating the entity:

```yaml
description: "Update Vartastorage Sensor"
mode: single
trigger:
  - platform: time_pattern
    seconds: /1
condition: []
action:
  - service: homeassistant.update_entity
    target:
      entity_id: sensor.my_vartastorage_sensor
```

Reply:
> Personally, I would've used a configuration as follows (untested):
> 
> ```yaml
> sensor:
>   - platform: varta_storage
>     scan_interval: 1
> ```

Hmm.. unfortunately this works not. The config is valid but the sensor still update only in the interval hardcoded in const.py 

The automation approach also does not work. The automation gets triggered once a second - but the entity_id configured in the action section does not get updated more faster then the configured 10 seconds.

Is there maybe a change needed to the code in const.py to allow the scan_interval to be overwritten by values in configuration.yaml?

Reply:
That's unfortunate. Personally, I have no experience with the use of `DataUpdateCoordinator`, but the SMA integration has a similar mechanic to make to update interval configurable: https://github.com/home-assistant/core/blob/dev/homeassistant/components/sma/__init__.py#L88

Reply:
Thats a great tip, thank you ! I'll loo into that and try to adjust the code of the integration accordingly.

Reply:
Thanks again Santober. I now changed this as described below:
- The default scan interval will be 10 seconds
- The user can configure to another value on the initial setup
- You also will be able to modify this (and other parameters) by reconfiguring the integration (added an OptionFlowHandler)

Reply:
: